### PR TITLE
Change tags to skills

### LINKED ISSUE #42: Convert tags into skills, add indexing to tags, sort alphabetically

### PR DESCRIPTION
**Pull request by linustws**
Closes #42.
- Changed all occurences of tags to skills.
- Replaced anything related to tags i.e., husband, friend, neighbour, colleagues... to skills like python, java, etc.
- Added default skills to the persons in SampleDataUtil.java, the data that it auto loads if the data file is deleted.

Also, I realised they used puml files to create the diagrams for the DG, when changing tags to skills in the puml files (docs/diagrams/BetterModelClassDiagram.puml, docs/diagrams/ModelClassDiagram.puml, docs/diagrams/StorageClassDiagram.puml), but I'm not sure how they converted them into png files. But we can use this method to create diagrams for our DG.

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -219,13 +219,13 @@ _Details coming soon ..._
...
 | Find	name          | `find NAME [MORENAMES]`<br />e.g., `find Bob Sam Pete`                                                                                                                                                                                                 |
 | Find module        | `findm MODCODE [MOREMODCODES]`<br />e.g., `findm cs2013t cs2109s cs2107`                                                                                                                                                                               |
-| Find tags          | `findt TAG [MORETAGS]`<br />e.g., `findt java python`                                                                                                                                                                                                  |
+| Find skills          | `findt SKILL [MORESKILLS]`<br />e.g., `findt java python`                                                                                                                                                                                                  |
```
Comment (harin0826) [inline docs/UserGuide.md:228]: Should `findt` be implemented as `finds`?

File: README.md [hunk 2]
```diff
@@ -20,21 +20,21 @@ Use the available commands to manage your contacts and connect with students in
...
+Add	| `add n/NAME y/YEAR c/COURSE e/EMAIL [l/LINKEDIN] [g/GITHUB] [m/YEARTAKEN MODCODE]... [s/SKILL]...`<br />e.g., `add n/Bob Sim y/2 c/com sci e/e0823741@nus.edu l/linkedin.com/in/bom-sim-086g93847/ m/ay2223s2 cs2103t m/ay2223s2 cs2101 s/python s/java`
 View contact | `view INDEX`<br />e.g., `view 3`
 View tab | `view C/M/T`<br />e.g., `view c`, `view m` or `view t`
-Edit | `edit [n/NAME] [y/YEAR] [c/COURSE] [e/EMAIL] [l/LINKEDIN] [g/GITHUB] [m/MODINDEX YEARTAKEN MODCODE]... [t/TAG]...`<br />e.g., `edit n/Bob Lim m/2 ay2223s2 cs2109s`
+Edit | `edit [n/NAME] [y/YEAR] [c/COURSE] [e/EMAIL] [l/LINKEDIN] [g/GITHUB] [m/MODINDEX YEARTAKEN MODCODE]... [s/SKILL]...`<br />e.g., `edit n/Bob Lim m/2 ay2223s2 cs2109s`
 Find	name | `find NAME [MORENAMES]`<br />e.g., `find Bob Sam Pete`
 Find module | `findm MODCODE [MOREMODCODES]`<br />e.g., `findm cs2013t cs2109s cs2107`
-Find tags | `findt TAG [MORETAGS]`<br />e.g., `findt java python` 
+Find skills | `findt SKILL [MORESKILLS]`<br />e.g., `findt java python`
```
Comment (harin0826) [inline README.md:29]: Should `findt` be implemented as `finds`?

**Review by harin0826 — APPROVED** [review]
Thank you for all the refactoring! Changing existing tags to skills under all Util classes must have been tedious. Other than small change suggestion in documentation for not yet implemented command for finding person with skills, I did not notice much errors in refactoring.

File: README.md (hunk 2 above)
Comment (linustws) [reply README.md]: Fixed, thanks for pointing that out 👍

File: docs/UserGuide.md (hunk 1 above)
Comment (linustws) [reply docs/UserGuide.md]: Fixed, thanks for pointing that out 👍

**Pull request merged**